source/scene.js

```diff
@@ -32,7 +32,64 @@ this.initialize = function() {
     this.initializeActiveCamera( this.gameCam.camera );
     this.setUpCameraListener();
     this.setUpRoverListeners();
-    this.gameCam.setCameraTarget( this.player.rover );
+    this.future( 3 ).applicationLoaded();
+}
+
+this.applicationLoaded = function() {
+    this.applicationState = "menu";
+}
+
+this.setApplicationState = function( state ) {
+    switch ( state ) {
+        case "loading":
+            break;
+        case "menu":
+            this.mainMenu.visible = true;
+            this.mainMenu.future( 0 ).setup();
+            // TODO: Consolidate game nodes
+            this.environment.visible = false;
+            this.player.visible = false;
+            this.airDust.visible = false;
+            this.smoke1.visible = false;
+            this.smoke2.visible = false;
+            this.smoke3.visible = false;
+            this.backdrop.visible = false;
+            this.sunLight.visible = false;
+            this.envLight.visible = false;
+            this.pickups.visible = false;
```

Comment:
Since you appear to be setting all of these to true/false, does it make sense to extract this out to a function somewhere or something?  Might be handy if we ever wanted to do the same thing somewhere else (like when loading a new area).  Just a thought...